### PR TITLE
CSP-955 OuterAPI EMP GetMemberProfileWithPreferences

### PR DESCRIPTION
**Pull request by YuvrajPatil-DFE**

### CODE REVIEW

**Review by MarkFCain — APPROVED** [review]

**Review by ManishAgarwalDfE — APPROVED** [review]

**Review by ManishAgarwalDfE — DISMISSED** [review]

File: src/EmployerAan/SFA.DAS.EmployerAan/Models/ApprenticeshipDetail.cs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+﻿namespace SFA.DAS.EmployerAan.Models;
+
+public class Apprenticeship
+{
+    public int ActiveApprenticesCount { get; set; }
+    public IEnumerable<string> Sectors { get; set; } = Enumerable.Empty<string>();
```
Comment (DevenKShah) [inline src/EmployerAan/SFA.DAS.EmployerAan/Models/ApprenticeshipDetail.cs:6]: Sectors exist in here as well as in ApprenticeshipsFilterValues, is that correct?

File: src/EmployerAan/SFA.DAS.EmployerAan/Application/MemberProfiles/Queries/GetMemberProfileWithPreferences/GetMemberProfileWithPreferencesQueryHandler.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    {
+        var responseMemberProfileWithPreferenceTask = _apiClient.GetMemberProfileWithPreferences(request.MemberId, request.RequestedByMemberId, request.IsPublicView, cancellationToken);
+        var responseMemberTask = _apiClient.GetMember(request.MemberId, cancellationToken);
+        var responseRegionsTask = _apiClient.GetRegions(cancellationToken);
+
+        await Task.WhenAll(responseMemberProfileWithPreferenceTask, responseMemberTask, responseRegionsTask);
+
+        GetMemberProfileWithPreferencesQueryResult result = new();
+
+        var outputMemberProfileWithPreferenceTask = responseMemberProfileWithPreferenceTask.Result;
```
Comment (DevenKShah) [inline src/EmployerAan/SFA.DAS.EmployerAan/Application/MemberProfiles/Queries/GetMemberProfileWithPreferences/GetMemberProfileWithPreferencesQueryHandler.cs:26]: variable should not be suffixed with `Task` as it is the result of the API call. Just remove `Task` or I suggest `memberProfileWithPreferences` which can be suffixed with `Response` or `Result`

File: src/EmployerAan/SFA.DAS.EmployerAan.Api/Controllers/MemberProfilesController.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
+﻿using MediatR;
+using Microsoft.AspNetCore.Mvc;
+using SFA.DAS.EmployerAan.Application.Employer.Queries.GetEmployerMemberSummary;
+using SFA.DAS.EmployerAan.Application.MemberProfiles.Queries.GetMemberProfileWithPreferences;
+using SFA.DAS.EmployerAan.Infrastructure;
+using SFA.DAS.EmployerAan.Models;
+
+namespace SFA.DAS.EmployerAan.Api.Controllers;
+
+[Route("Members")]
```
Comment (DevenKShah) [inline src/EmployerAan/SFA.DAS.EmployerAan.Api/Controllers/MemberProfilesController.cs:10]: All the routes should be in lower case

Comment (DevenKShah) [reply src/EmployerAan/SFA.DAS.EmployerAan.Api/Controllers/MemberProfilesController.cs:10]: This is missing ApiController attribute

File: src/EmployerAan/SFA.DAS.EmployerAan.Api.UnitTests/Controllers/MemberProfiles/MemberProfilesControllerTests.cs [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        CancellationToken cancellationToken)
+    {
+        memberProfileWithPreferencesQueryResult.AccountId = accountId;
+        GetEmployerMemberSummaryQuery employerMemberSummaryQuery = new(accountId);
+        GetMemberProfileWithPreferencesModel response = new(memberProfileWithPreferencesQueryResult, employerMemberSummaryQueryResult);
+
+        mockMediator.Setup(m => m.Send(It.IsAny<GetMemberProfileWithPreferencesQuery>(), cancellationToken)).ReturnsAsync(memberProfileWithPreferencesQueryResult);
+        mockMediator.Setup(m => m.Send(employerMemberSummaryQuery, cancellationToken)).ReturnsAsync(employerMemberSummaryQueryResult);
+
+        var controller = new MemberProfilesController(mockMediator.Object);
```
Comment (DevenKShah) [inline src/EmployerAan/SFA.DAS.EmployerAan.Api.UnitTests/Controllers/MemberProfiles/MemberProfilesControllerTests.cs:36]: This should be renamed to `sut`

**Review by DevenKShah — APPROVED** [review]
Few nagging comments

File: src/EmployerAan/SFA.DAS.EmployerAan.Api/Controllers/MemberProfilesController.cs (hunk 3 above)
Comment (YuvrajPatil-DFE) [reply src/EmployerAan/SFA.DAS.EmployerAan.Api/Controllers/MemberProfilesController.cs:10]: This is fixed in codereview ticket, but will fix it here as well.

File: src/EmployerAan/SFA.DAS.EmployerAan/Models/ApprenticeshipDetail.cs (hunk 1 above)
Comment (YuvrajPatil-DFE) [reply src/EmployerAan/SFA.DAS.EmployerAan/Models/ApprenticeshipDetail.cs:6]: Yes

**Review by DevenKShah — APPROVED** [review]

**Pull request merged**